src/rgw/rgw_rest_s3.h

```diff
@@ -113,7 +112,7 @@ class RGWListBuckets_ObjStore_S3 : public RGWListBuckets_ObjStore {
     return 0;
   }
   void send_response_begin(bool has_buckets) override;
-  void send_response_data(RGWUserBuckets& buckets) override;
+  void send_response_data(rgw::sal::RGWBucketList& buckets) override;
```

Comment:
at the very least 
rgw::sal::Bucket is preferable to RGWSalBucket--but again, I think I'm liking rgw::storage::Bucket

Reply:
Like I said, RGWSalBucket is temporary, and will be RGWBucket.